cirq/google/gate_sets.py

```diff
@@ -12,116 +12,167 @@
-                                               serialized_name='half_turns',
-                                               serialized_type=float,
-                                               gate_getter='exponent')
-                                       ]),
-        op_serializer.GateOpSerializer(
-            gate_type=ops.MeasurementGate,
-            serialized_gate_id='meas',
-            args=[
-                op_serializer.SerializingArg(
-                    serialized_name='key',
-                    serialized_type=str,
-                    gate_getter=protocols.measurement_key),
-                op_serializer.SerializingArg(serialized_name='invert_mask',
-                                             serialized_type=List[bool],
-                                             gate_getter='invert_mask')
-            ])
+        _SYC_SERIALIZER,
+        *SINGLE_QUBIT_SERIALIZERS,
+        *SINGLE_QUBIT_HALF_PI_SERIALIZERS,
+        MEASUREMENT_SERIALIZER,
     ],
     deserializers=[
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='exp_w',
-            gate_constructor=ops.PhasedXPowGate,
-            args=[
-                op_deserializer.DeserializingArg(
-                    serialized_name='axis_half_turns',
-                    constructor_arg_name='phase_exponent'),
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='exp_z',
-            gate_constructor=ops.ZPowGate,
-            args=[
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='exp_11',
-            gate_constructor=ops.CZPowGate,
-            args=[
-                op_deserializer.DeserializingArg(
-                    serialized_name='half_turns',
-                    constructor_arg_name='exponent')
-            ]),
-        op_deserializer.GateOpDeserializer(
-            serialized_gate_id='meas',
-            gate_constructor=ops.MeasurementGate,
-            args=[
-                op_deserializer.DeserializingArg(serialized_name='key',
-                                                 constructor_arg_name='key'),
-                op_deserializer.DeserializingArg(
-                    serialized_name='invert_mask',
-                    constructor_arg_name='invert_mask',
-                    value_func=lambda x: tuple(cast(list, x)))
-            ],
-            num_qubits_param='num_qubits'),
-    ])
+        _SYC_DESERIALIZER,
+        *SINGLE_QUBIT_DESERIALIZERS,
+        *SINGLE_QUBIT_HALF_PI_DESERIALIZERS,
+        MEASUREMENT_DESERIALIZER,
+    ],
+)
+
+# The xmon gate set.
+XMON: serializable_gate_set.SerializableGateSet = (
```

Comment:
why is this type annotation needed? why isn't it needed for syc gate set

Reply:
I am not totally sure why it is needed.  I think something in the import order changed and mypy started complaining about not knowing the type of XMON, which is used as the default in cirq/google/engine.py.  This is the only way I could get it to stop.